stix2elevator/convert_cybox.py

```diff
@@ -459,6 +469,96 @@ def create_domain_name_object(dn):
     return cybox_dict
 
 
+def split_into_requests_and_responses(req_resp_list):
+    requests = []
+    responses = []
+    for r in req_resp_list:
+        if r.http_client_request:
+            requests.append(r.http_client_request)
+        if r.http_server_response:
+            responses.append(r.http_server_response)
+    return requests, responses
+
+
+def convert_http_session(session):
+    if session.http_request_response:
+        requests, responses = split_into_requests_and_responses(session.http_request_response)
+        if len(responses) != 0:
+            warn("HTTPServerResponse type is not supported in STIX 2.0", 429)
+        if len(requests) >= 1:
+            cybox_traffic = {"type": "network-traffic"}
+            cybox_traffic["extensions"] = {"http-request-ext": convert_http_client_request(requests[0])}
+            if len(requests) > 1:
+                warn("Only HTTP_Request_Response used for http-request-ext, using first value", 512)
+            return cybox_traffic
+
+
+def create_icmp_extension(icmp_header):
+    imcp_extension = {}
+    if icmp_header.type_:
+        imcp_extension["icmp_type_hex"] = icmp_header.type_.value
+    if icmp_header.code:
+        imcp_extension["icmp_code_hex"] = icmp_header.code.value
+    if icmp_header.checksum:
+        warn("ICMP_Packet/Checksum content not supported in STIX 2.0", 424)
+    return imcp_extension
+
+
+def convert_network_packet(packet):
+    if packet.internet_layer:
+        internet_layer = packet.internet_layer
+        if internet_layer.ipv4 or internet_layer.ipv6:
+            warn("Internet_Layer/IP_Packet content not supported in STIX 2.0", 424)
+        else:
+            if internet_layer.icmpv4:
+                icmp_header = internet_layer.icmpv4.icmpv4_header
+            elif internet_layer.icmpv6:
+                icmp_header = internet_layer.icmpv6.icmpv6_header
+            else:
+                return None
+            cybox_traffic = {"type": "network-traffic"}
+            cybox_traffic["extensions"] = {"icmp-ext": create_icmp_extension(icmp_header)}
+            return cybox_traffic
+
+
+def convert_socket_options(options):
+    socket_options = {}
+    for prop_name in SOCKET_OPTIONS:
+        if getattr(options, prop_name):
+            socket_options[prop_name.upper()] = getattr(options, prop_name)
+    return socket_options
+
+
+def convert_network_socket(socket):
+    cybox_traffic = {"type": "network-traffic"}
+    socket_extension = {}
+    if socket.is_blocking:
+        socket_extension["is_blocking"] = socket.is_blocking
+    if socket.is_listening:
+        socket_extension["is_listening"] = socket.is_listening
+    if socket.address_family:
+        if socket.address_family in ADDRESS_FAMILY_ENUMERATION:
+            socket_extension["address_family"] = socket.address_family.value
+        else:
+            warn("%s in is not a member of the %s enumeration", 627, socket.address_family, "address family")
```

Comment:
does the warning message need to include an identifier after "in"?

Reply:
Good catch.  Unfortunately, it is difficult to know the context here, so I'm removing the "in".
In a similar situation in convert_pattern, I passed in the id of the observable, but Greg thought that just passing the id for message purposes wasn't the best.